code/script.js

```diff
@@ -222,96 +227,136 @@ const generateBoard = () => {
+      }
+      else {
+        alert(
+        `No! The person does not have a ${value}! Remove everyone with a ${value}.`)
+      charactersInPlay = charactersInPlay.filter((person) => !person[category])
+      }
+  // Hair
+  } else if (category === "hair") {
+      if (keep) {
+        alert(
+        `Yes! The person has ${value} hair! Keep everyone with ${value} hair.`)
+        charactersInPlay = charactersInPlay.filter(
+          (person) => person[category] === value)
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+        alert(
+        `No! The person does not have ${value} hair! Remove everyone with ${value} hair.`)
+        charactersInPlay = charactersInPlay.filter(
+          (person) => person[category] !== value)}
+    // Eyes
+    } else if (category === "eyes") {
+      if (keep) {
+        alert(
+          `Yes! The person has ${value} eyes! Keep everyone with ${value} eyes.`)
+          charactersInPlay = charactersInPlay.filter(
+            (person) => person[category] === value)
+      }
+      else {
+        alert(
+          `No! The person does not have ${value} eyes! Remove everyone with ${value} eyes.`)
+          charactersInPlay = charactersInPlay.filter(
+            (person) => person[category] !== value)
+      }
     }
+    generateBoard()
   }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
   // Invoke a function to redraw the board with the remaining people.
-}
+ 
+
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
+// Prompt user to confirm they want to make a guess.
 const guess = (personToConfirm) => {
   // store the interaction from the player in a variable.
+  const confirmGuess = confirm(`Are you sure you want to guess?`)
   // remember the confirm() ?
+  if (confirmGuess) checkMyGuess(personToConfirm)
   // If the player wants to guess, invoke the checkMyGuess function.
 }
 
 // If you confirm, this function is invoked
-const checkMyGuess = (personToCheck) => {
+const checkMyGuess = (personToConfirm) => {
+  if (personToConfirm === secret.name) {
+    winOrLoseText.innerHTML = `It was indeed ${secret.name}! Well done!`
+  } else {
+    winOrLoseText.innerHTML = `Oh no, that was not the person we were looking for...The correct guess is ${secret.name}!`
+  }
+  winOrLose.style.display = "flex"
```

Comment:
Nice that you added who the secret person is here!

Reply:
Yes thank you, it's a nice touch isn't it!